lib/gui/components/svg-icon.js

```diff
@@ -65,12 +67,13 @@ class SVGIcon extends react.Component {
     const width = this.props.width || DEFAULT_SIZE;
     const height = this.props.height || DEFAULT_SIZE;
 
-    const parser = new window.DOMParser();
-    const doc = parser.parseFromString(contents, 'image/svg+xml');
+    const img = document.createElement('img');
+    const doc = domParser.parseFromString(contents, 'image/svg+xml');
+    const parserError = doc.querySelector('parsererror');
     const svg = doc.querySelector('svg');
+    const svgXml = svg && _.isNil(parserError) ? svg.outerHTML : '';
```

Comment:
Since we're going to introduce a new attribute for embedded contents, maybe we can prioritize the embedded SVG attribute and use the path as a fallback?